crypto/src/aggregate.rs

```diff
@@ -0,0 +1,40 @@
+use std::collections::HashMap;
+
+use blsttc::{Error, PublicKeySet, SignatureShare};
+
+use crate::keypair::PublicKey;
+use crate::signature::Signature;
+
+pub struct Aggregator(PublicKeySet);
+
+impl Aggregator {
+    pub fn new(raw_bytes: Vec<u8>) -> Self {
+        Self(PublicKeySet::from_bytes(raw_bytes).unwrap())
+    }
+
+    pub fn pubkey(&self) -> PublicKey {
+        PublicKey::new(self.0.public_key().to_bytes())
+    }
+
+    pub fn aggregate_signatures(
```

Comment:
Can we accept a list of Signature instead of HashMap?

Reply:
Since `id` is crucial for aggregating signatures, I used a hashmap to store `id` as the key. Using a list would prevent including the `id` information, leading to a risk of elements being misplaced in their original indexes. To address this, I added a test called `test_aggregate_signatures_wrong_id`.

Reply:
I see, so the order of partial signatures must correspond to that of keypairs. It seems it requires us to assign `id` for each node and use it for the ordering.